src/main/scala/edu/ie3/mobsim/MobilitySimulator.scala

```diff
@@ -574,6 +582,20 @@ final class MobilitySimulator(
 
     timeUntilNextEvent
   }
+
+  private def updateElectricVehicles(movements: Seq[Movement]): Unit = {
+    val movementMap: Map[UUID, ElectricVehicle] = movements.map { movement =>
+      movement.ev.uuid -> movement.ev
+    }.toMap
+
+    electricVehicles = electricVehicles.map { ev =>
+      if (movementMap.contains(ev.uuid)) {
+        movementMap(ev.uuid)
+      } else {
+        ev
+      }
```

Comment:
This can be shortened:
```suggestion
      movementMap.getOrElse(ev.uuid, ev)
```